pkg/api/converterfromagentpoolonlyapi.go

```diff
@@ -130,12 +130,37 @@ func convertResourcePurchasePlanToV20180331AgentPoolOnly(api *ResourcePurchasePl
 	v20180331.Publisher = api.Publisher
 }
 
+func convertKubernetesConfigToEnableRBACV20180331AgentPoolOnly(kc *KubernetesConfig) bool {
+	if kc == nil {
+		return false
+	}
+	// Only if both KubernetesConfig.EnableRbac and KubernetesConfig.EnableSecureKubelet true
+	// We think EnableRBAC is on
+	enableRBAC := true
+	if kc.EnableRbac == nil {
+		enableRBAC = enableRBAC && false
+	} else {
+		enableRBAC = enableRBAC && *kc.EnableRbac
+	}
+
+	if kc.EnableSecureKubelet == nil {
```

Comment:
IMO if the code is `convertertoagentpoolonlyapi.go` is written as expected then we shouldn't need to check for `EnableSecureKubelet` state. Checking `enableRBAC` flag state should suffice.

Reply:
The way we want to make it, is to have `EnableSecureKubelet` also enabled when use specify enableRBAC. I should make it clear in the enableRBAC flag description. Check the email thread I included you.

Reply:
Yes, agreed. We are on the same page regarding:
The way we want to make it, is to have EnableSecureKubelet also enabled when use specify enableRBAC.

I am suggesting that if rbac was enabled in API model then `EnableSecureKubelet` would have been set to true. So when converting it back to versioned API model state of  `EnableSecureKubelet` should not need to be validated. Another way to put it is rbac can still be technically enabled if even  `EnableSecureKubelet` was set to false.

Reply:
why `enableRBAC && false` when it must be `false`?

Reply:
what email thread?

> Check the email thread I included you.